cmd/compute-domain-controller/daemonset.go

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) 2025 NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+NumberReady * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+	"text/template"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	nvapi "github.com/NVIDIA/k8s-dra-driver-gpu/api/nvidia.com/resource/v1beta1"
+)
+
+const (
+	DaemonSetTemplatePath = "/templates/compute-domain-daemon.tmpl.yaml"
+)
+
+type DaemonSetTemplateData struct {
+	Namespace                 string
+	GenerateName              string
+	Finalizer                 string
+	ComputeDomainLabelKey     string
+	ComputeDomainLabelValue   types.UID
+	ResourceClaimTemplateName string
+}
+
+type DaemonSetManager struct {
+	sync.Mutex
```

Comment:
I wondered what this does and how this works. Found https://stackoverflow.com/a/44950096/145400. Nice syntactical sugar.